bloom-housing/bloom
# feat: uptake toast component from ui-seeds

# Pull Request Template

## Issue Overview

This PR addresses #3434


## Description

Pulls in the toast component from ui-seeds and replaces all places the ui-component version of toast was used with the new one.

Notes:
- In an attempt to standardize how the toast is used and displayed, a message context was added. This allows us to set a message anywhere on the site and it will appear on the toast in the same place of the page.
  - The places where `<Toast>` is placed in the app is reduced
- The users page had a somewhat big rewrite in order to prevent unnecessary re-renders. Previously after a new user was added and the modal closed, a call to the backend is triggered which triggered the whole page to re-render. This is is because the api call was getting done at too high of a level. The new format makes it so only the users table area of the page is re-rendered
  - This follows best practices but also prevents the toast from getting rendered and then immediately disappearing after new api call is done

## How Can This Be Tested/Reviewed?

There are many Toasts throughout the site
- Invite a new user
- Delete a User
- Update a listing
- Export CSV on listings and users pages
- ~Confirm account~

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
